lib/inspec/cli.rb

```diff
@@ -99,36 +99,32 @@ def check(path) # rubocop:disable Metrics/AbcSize
       puts JSON.generate(result)
     else
       %w{location profile controls timestamp valid}.each do |item|
-        puts format('%-12s %s', item.to_s.capitalize + ':',
-                    mark_text(result[:summary][item.to_sym]))
+        prepared_string = format('%-12s %s',
+                                 (item.to_s.capitalize + ':'),
+                                 result[:summary][item.to_sym])
+        ui.plain_line(prepared_string)
       end
       puts
 
       if result[:errors].empty? and result[:warnings].empty?
-        puts 'No errors or warnings'
+        ui.plain_line('No errors or warnings')
       else
-        red    = "\033[31m"
-        yellow = "\033[33m"
-        rst    = "\033[0m"
-
         item_msg = lambda { |item|
           pos = [item[:file], item[:line], item[:column]].compact.join(':')
           pos.empty? ? item[:msg] : pos + ': ' + item[:msg]
         }
-        result[:errors].each do |item|
-          puts "#{red}  ✖  #{item_msg.call(item)}#{rst}"
-        end
-        result[:warnings].each do |item|
-          puts "#{yellow}  !  #{item_msg.call(item)}#{rst}"
-        end
+
+        result[:errors].each { |item| ui.red " #{Inspec::UI::GLYPHS[:script_x]}  #{item_msg.call(item)}\n" }
+        result[:warnings].each { |item| ui.yellow " !  #{item_msg.call(item)}\n" }
 
         puts
-        puts format('Summary:     %s%d errors%s, %s%d warnings%s',
-                    red, result[:errors].length, rst,
-                    yellow, result[:warnings].length, rst)
+
+        errors = ui.red(result[:errors].length.to_s + ' errors', print: false)
+        warnings = ui.yellow(result[:warnings].length.to_s + ' warnings', print: false)
```

Comment:
In the same sense, whenever you see `x.to_s + y`, that can be much better expressed with interpolation or `String#%`:

```ruby
# using interpolation
errors = ui.red("#{result[:errors].length} errors", print: false)
warnings = ui.yellow("#{result[:warnings].length} warnings", print: false)

# using String#% (aka format)
errors = ui.red("%d errors" % [result[:errors].length], print: false)
warnings = ui.yellow("%d warnings" % [result[:warnings].length], print: false)
```

Totally minor in this context tho.